api/files/views.py

```diff
@@ -35,7 +35,249 @@ def get_file(self, check_permissions=True):
+    the behind-the-scenes moving, uploading, deleting, etc. of files and folders is actually be handled for us by a
+    nifty piece of software called [WaterButler](https://github.com/CenterForOpenScience/waterbutler).  WaterButler lets
+    us interact with files stored on different cloud storage platforms through a consistent API.  However, it uses
+    different conventions for requests, responses, and URL-building, so pay close attention to the documentation for
+    [actions](#actions).  The exceptions are the "Get Info" and "Checkout" actions, which are OSF-centric.
+
+    If your file or folder is stored on a non-OSF provider, the file metadata will be retrieved if not already present
+    in the system.  Otherwise, the cached metadata will be presented.  To force metadata retrieval, you'll need to issue
+    a GET request to the Node Files List endpoint.
+
+    Both files and folders are available through the Files API and are distinguished by the `kind` attribute ("file" for
+    files, "folder" for folders).  Not all actions and relationships are relevent to both files and folders, so the
+    applicable types are listed by each heading.
+
+    ###Waterbutler Entities
+
+    When an action is performed against a WaterButler endpoint, it will generally respond with a file entity, a folder
+    entity, or no content.
+
+    ####File Entity
+
+        name          type       description
+        -------------------------------------------------------------------------
+        name          string     name of the new folder
+        path          string     OSF path/id of the file
+        materialized  string     the full path of the folder relative to the storage root
+        kind          string     "file"
+        etag          string     etag - http caching identifier
+        modified      timestamp  last modified timestamp - format depends on provider
+        contentType   string     null if provider="osfstorage", else MIME-type
+        provider      string     id of provider e.g. "osfstorage", "s3", "googledrive"
+        size          integer    size of file in bytes
+        extra         object
+          version     integer    version number of file. will be 1 on initial upload
+          downloads   integer    count of the number times the file has been downloaded
+          hashes      object
+            md5       string     md5 hash of file
+            sha256    string     SHA-256 hash of file
+
+    ####Folder Entity
+
+        name          type    description
+        ----------------------------------------------------------------------
+        name          string  name of the new folder
+        path          string  OSF path/id of the folder
+        materialized  string  the full path of the folder relative to the storage root
+        kind          string  "folder"
+        etag          string  etag - http caching identifier
```

Comment:
Maybe worth mentioning, but etag is the sha256 of uniquely identifying data for a file.
Note: diverges from the HTTP header etag as this value is not wrapped in double quotes, `{ETAG}` not `"{ETAG}"`